oneview/resource_volume.go

```diff
@@ -0,0 +1,336 @@
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceVolume() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVolumeCreate,
+		Read:   resourceVolumeRead,
+		Update: resourceVolumeUpdate,
+		Delete: resourceVolumeDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"category": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"device_volume_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"device_specific_attributes": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"copy_state": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"is_compressed": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"is_deduplicated": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"snapshot_pool_uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"properties": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"storage_pool": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"size": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"provisioning_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"data_transfer_limit": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"is_deduplicated": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"is_encrypted": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"is_pinned": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"is_compressed": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"data_protection_level": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"eTag": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"is_permanent": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"is_shareable": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"provisioned_capacity": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"provisioning_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"allocated_capacity": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"requesting_refresh": {
+				Type:     schema.TypeBool,
+				Computed: true,
```

Comment:
This attribute can be set while requesting for refresh, using a put call.

Reply:
this is not for updating volume. that's why not given as an editable option.